### PR TITLE
PoC: Do not look for `dashboardView` if GA4 is not connected

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #7116 

## Relevant technical choices

This PR prevents the `analytics` `isGA4DashboardView` selector from attempting to resolve the current Analytics dashboard view until it has been detected that GA4 is connected.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by aaemnnosttv — COMMENTED** [review]
This is what I was thinking of, although there are a few important bits to consider

- We should avoid showing the `SwitchedToGA4Banner` in a view-only context since this on-the-fly change in a view-only context would require refreshing permissions after changing the setting in order for GA4 to be viewable. I don't think we're currently restricting the visibility of this banner which we should only show if the user has the necessary capability to manage settings.
- The experience for other view-only users in the event of a change of this setting. Probably rare/unlikely, but imagine being a view only user looking at one version of the dashboard while another admin changes the dashboard view. Then if you were to do something to trigger another report request, like click a slice of the pie chart or change the date range, it would error.

File: includes/Modules/Analytics.php [hunk 1]
```diff
@@ -158,6 +158,44 @@ function ( $data ) {
 			}
 		);
 
+		// Conditionally allow access to either Analytics module based on current dashboard view.
+		if ( Feature_Flags::enabled( 'ga4Reporting' ) ) {
+			add_filter(
+				'user_has_cap',
+				function( array $allcaps, $caps, $args ) {
+					$module_slug = $args[2];
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply includes/Modules/Analytics.php]: Ah, nice catch, updated, thank you!

File: includes/Modules/Analytics.php [hunk 2]
```diff
@@ -158,6 +158,44 @@ function ( $data ) {
...
+
+					if (
+						Permissions::READ_SHARED_MODULE_DATA === $args[0] &&
+						in_array(
+							$module_slug,
+							array( self::MODULE_SLUG, Analytics_4::MODULE_SLUG ),
+							true
+						)
+					) {
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply includes/Modules/Analytics.php]: Updated, thank you!

**Comment by nfmohit** [comment]
Thank you for the very kind feedback, @aaemnnosttv! I have addressed the code specific feedback.

> * We should avoid showing the `SwitchedToGA4Banner` in a view-only context since this on-the-fly change in a view-only context would require refreshing permissions after changing the setting in order for GA4 to be viewable. I don't think we're currently restricting the visibility of this banner which we should only show if the user has the necessary capability to manage settings.

In the `BannerNotifications` component, we're only showing the `ZeroDataStateNotifications` in the view-only context. All other notifications including `SwitchedToGA4Banner` are not shown in the view-only context, according to the following:

https://github.com/google/site-kit-wp/blob/582bcced87d1db3907a6034e4d7b9c8366f0289b/assets/js/components/notifications/BannerNotifications.js#L100-L102

Do you think that is sufficient?

> * The experience for other view-only users in the event of a change of this setting. Probably rare/unlikely, but imagine being a view only user looking at one version of the dashboard while another admin changes the dashboard view. Then if you were to do something to trigger another report request, like click a slice of the pie chart or change the date range, it would error.

This is a very interesting scenario. Should we somehow trigger a refresh on the view-only dashboard if it detects that the dashboard view has changed? I'm open to ideas. Also, do you think it'd be wiser to address this in a follow-up enhancement issue once we've rolled out a fix for the originally reported issue?

Thank you!

**Comment by aaemnnosttv** [comment]
> In the `BannerNotifications` component, we're only showing the `ZeroDataStateNotifications` in the view-only context. All other notifications including `SwitchedToGA4Banner` are not shown in the view-only context, according to the following:

Oh great, I must have assumed it was. Thanks for pointing that out 👍 

> This is a very interesting scenario. Should we somehow trigger a refresh on the view-only dashboard if it detects that the dashboard view has changed? I'm open to ideas. Also, do you think it'd be wiser to address this in a follow-up enhancement issue once we've rolled out a fix for the originally reported issue?

That sounds reasonable considering this is probably a very unlikely edge case. I was thinking we could perhaps return a specific error (or maybe the current error would be sufficient) and then we could trigger either a page refresh or simply reload the user's permissions and GA settings (I think that would do it 🤔). This could indeed be handled afterwards though.

The main concern I had was around similar implications with the `SwitchGA4DashboardViewNotification` (not `SwitchedToGA4Banner`, I often mix these) because this is the only place where we switch the dashboard view _on the dashboard_, but this should be fine so long as it doesn't happen in a shared context where the capability would need refreshing.

The updates to the PR look better here, I'll leave another review but this looks good to continue with 👍 Thanks @nfmohit !

**Review by aaemnnosttv — COMMENTED** [review]
Thanks @nfmohit – a few more thoughts here 😄

File: includes/Modules/Analytics.php [hunk 3]
```diff
@@ -158,6 +158,46 @@ function ( $data ) {
 			}
 		);
 
+		// Conditionally allow access to either Analytics module based on current dashboard view.
+		if ( Feature_Flags::enabled( 'ga4Reporting' ) ) {
+			add_filter(
+				'user_has_cap',
+				function( array $allcaps, $caps, $args ) {
```
[the remark this answers is not in the document]
Comment (nfmohit) [reply includes/Modules/Analytics.php]: I completely agree and that makes sense. I've removed this change from the PR. Thank you @aaemnnosttv!

**Pull request merged**